openage/convert/processor/export/generate_manifest_hashes.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020-2020 the openage authors. See copying.md for legal info.
+"""
+Provides functions for traversing a directory and
+generating hash values for all the items inside.
+"""
+
+import hashlib
+import os
+
+
+def bfs_directory(root):
+    """
+    Traverse the given directory with breadth-first way.
+
+    :param root: The directory to traverse.
+    :type root: ...util.fslike.path.Path
+    """
+
+    # initiate from the root directory
+    dirs = [root]
+
+    while dirs:
+        next_level = []
+
+        for directory in dirs:
+            for item in directory.iterdir():
+                item_path = directory.joinpath(item)
```

Comment:
This line causes a runtime error. Creating `item_path` fails because `item` is already a `Path` object.

You do not need `item_path` or the `.joinpath()` command here. Just remove this line and change all references of `item_path` to `item`. Then it runs fine.